9.0: FEATURE: Improve ESCR Compatibility - Part 1

ahaeslich opened the pull request:
- Use `Neos.Node.isLive(...)` instead of `this.node.context.workspace.name == 'live'`
- Update `Neos.Seo:LangAttribute.fusion` to get correct language based on new node properties
- Use new `Neos.Dimension` helper for `AlternateLanguageLinks`

Tested features:

- TitleTag
- MetaRobots, MetaKeyword, MetaDescription
- Lang attribute at HTML tag
- Canonical and alternative links
- Social meta data

Relates: #171 
Relates: neos/neos-development-collection#4074
Relates: neos/neos-development-collection#4085

ahaeslich committed:
FEATURE: Use new eel helper `Neos.Node.isLive(...)`

- Use `Neos.Node.isLive(...)` instead of `this.node.context.workspace.name == 'live'`
- Update `Neos.Seo:LangAttribute.fusion` to get correct language based on new node properties

Relates: neos/neos-development-collection#4074

ahaeslich committed:
FEATURE: Use new `Neos.Dimension` helper for `AlternateLanguageLinks`

ahaeslich committed:
TASK: Refactor `Neos.Seo:LangAttribute` and `Neos.Seo:AlternateLanguageLinks` based on changes to `Neos.Dimension` helper

nezaniel commented on Resources/Private/Fusion/Metadata/AlternateLanguageLinks.fusion:
```diff
@@ -1,34 +1,34 @@
-            @if.notExcluded = ${!props.excludedPresets || Array.indexOf(props.excludedPresets, this.dimensionValue) == -1}
+            @if.indexingAllowed = ${q(this.node).property('metaRobotsNoindex') != true}
+            @if.notExcluded = ${!props.excludedPresets || Array.indexOf(props.excludedPresets, dimensionValue.value) == -1}
 
```
Check for null node

nezaniel commented on Resources/Private/Fusion/Metadata/LangAttribute.fusion:
```diff
@@ -1,5 +1,8 @@
-    value = ${String.replace(documentNode.subgraphIdentity.dimensionSpacePoint.coordinates.language, '_', '-')}
-    @if.languageDimensionExists = ${documentNode.subgraphIdentity.dimensionSpacePoint.coordinates.language}
+    dimension = 'language'
+
```
separator prop

bwaidelich commented on Resources/Private/Fusion/Metadata/AlternateLanguageLinks.fusion:
```diff
@@ -1,29 +1,36 @@
     dimension = 'language'
-    defaultLocale = ${Configuration.setting('Neos.ContentRepository.contentDimensions.' + this.dimension + '.default')}
-    dimensionConfiguration = ${Configuration.setting('Neos.ContentRepository.contentDimensions.' + this.dimension)}
+    languageDimension = ${Neos.Dimension.all(this.node)[this.dimension]}
```
IMO we should try avoid array access (or other low level operators) if possible.
Couldn't we extend the helper such that the dimension can be specified like
```
languageDimension = ${Neos.Dimension.all(this.node, this.dimension)}
```
or add a separate helper (see https://github.com/neos/Neos.Demo/pull/160#discussion_r1156890517) ?

bwaidelich commented on Resources/Private/Fusion/Metadata/AlternateLanguageLinks.fusion:
```diff
@@ -1,29 +1,36 @@
     renderer = Neos.Fusion:Loop {
-        items = Neos.Neos:DimensionsMenuItems {
-            dimension = ${props.dimension}
+        items = ${props.languageDimension.values.values}
```
...and that helper should probably return the _values_.. `values.values` is rather confusing :)

ahaeslich replied:
Yes, I found both `values.values` or `values.interator` not a good solution 😉. That's why I started https://github.com/neos/Neos.Demo/pull/160#discussion_r1155327695

ahaeslich committed:
TASK: Adjust to changes in return type of `Neos.Dimension` helper

- Use `Neos.Dimension.allDimensionValues(site, 'language')`

Relates: https://github.com/neos/neos-development-collection/pull/4244
Relates: https://github.com/neos/neos-development-collection/pull/4249

ahaeslich committed:
TASK: Use `Neos.Site.findBySiteNode()` to get `site` object for `Neos.Seo:TitleTag`

mficzel reviewed (approved):
I like ... and tested with anke
